src/chat/tools/GetAzureActivityLog/GetAzureActivityLog.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtLMTool, IActionContext } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ActivityLogPromptType, GetAzureActivityLogInputSchema } from './GetAzureActivityLogInputSchema';
+import { convertActivityTreeToSimpleObjectArray, ConvertedActivityItem } from './convertActivityTree';
+
+export class GetAzureActivityLog<T extends GetAzureActivityLogInputSchema> implements AzExtLMTool<T> {
+    public async invoke(context: IActionContext, options: vscode.LanguageModelToolInvocationOptions<T>, _token: vscode.CancellationToken): Promise<vscode.LanguageModelToolResult> {
+        const convertedActivityItems: ConvertedActivityItem[] = await convertActivityTreeToSimpleObjectArray(context, options.input.treeId);
+
+        return {
+            content: [
+                convertedActivityItems.length ?
+                    constructLMToolResponse(JSON.stringify(convertedActivityItems), options.input.promptType ?? ActivityLogPromptType.Explain) :
+                    new vscode.LanguageModelTextPart('There is no activity data to analyze.')
+            ]
+        };
```

Comment:
IIRC, `vscode.LanguageModelToolResult` is a class, so we should use the constructor to create the return result.